### ISSUE TITLE
Rename Resource Descriptor class

### ISSUE THREAD

**Issue by nicholascar** [w3c/dx-prof#6]
The Resource Descriptor class' name may not convey its intention well. This name needs to be reconsidered with discussion about it here.

**Comment by kcoyle** [comment]
@makxdekkers, 

I think this is really a moot discussion due to the nature of the profiles ontology. Look again at the profiles ontology and you'll see that there are NO terms describing the profile itself or the resources, other than a small number of relationships between profiles and resources. Here are the properties associated with the main classes:

Profile:
hasProfile
hasResource
isProfileOf
isTransitiveProfileOf
hasToken

ResourceDescriptor:
hasArtifact
dct:conformsTo
dct:format
hasRole
isInheritedFrom

Compare that to how rich the descriptions of dcat:Catalog and dcat:Dataset and dcat:Distribution are. There is nothing of that nature in the profiles ontology. The properties given in @smrgeoinfo 's diagram are not included in the ontology, and there isn't much to tell us what information would live where.  It isn't even clear to me what properties one could include to describe a profile at the prof:Profile level because it's current a pretty vague concept, but I suspect a property set would be more sparse than one has in DCAT because what "makes up" the profile could vary greatly. In fact, we don't know much about what would make up a profile, if all "resources" would have the same author or institution, etc. None of this is specified in the ontology.

In any case, while @smrgeoinfo's model may appeal, it is not representative of what we are offering as a FPWD. If you do read that into the profiles ontology then you should present a proposal to the group and see what comes of it.

**Comment by kcoyle** [comment]
Yes, I was responding to Richard's model. In @smrgeoinfo 's model, ResourceDescriptor has only Role and the meat of the description of the resource is in the distribution/artifact. I question that for the case of profiles.

The only thing else I can say is that I do not believe that there will be _distributions_ for profiles if the definition is the same as it is in DCAT and if a profile = dataset:

Definition: | A  specific representation of a dataset. A dataset might be available in  several different forms, and these forms might comprise both different  serializations or different schematic arrangements of the same data. Examples of distributions include a CSV file, a netCDF  file, or a data-cube

There might be distributions for resources, but then resources become logically equivalent to datasets and the profile is a catalog. A profile, as I read profilesOnt, is an abstraction that is defined by one or more objects. Those objects are not for the most part in any way equivalent, so they are not distributions of the profile because they are not "the same data." 

I still see profiles as not being analogous to DCAT because DCAT has a "center" in the format of a dataset, and profiles have at best a named graph that is defined by its objects. The structure might be similar, but the concepts of dataset/distribution and profile/resource are not at all the same.

**Comment by nicholascar** [comment]
If we use a custom property like `prof:hasProfileObject` then we loose any speedup regarding easily understanding what the `prof:Profile` <-> `prof:ProfileObject` relationship is, although it may be able to be inferred from the class naming. 

I guess it depends what you think a 'profile' is. In the dummy DCAP profile I've made, https://github.com/CSIRO-enviro-informatics/csiro-epub-dcap, there is a PDF doc explaining things (Guidance) and Word form (`dct:hasFormat`) and constraints in RDF (not SHACL but DSP; equivalent to SHACL for this discussion's purposes). How would those all not be `dct:hasPart` from the main Profile object? They are all parts of it!

**Comment by nicholascar** [comment]
I'm using `ResourceDescriptor` as a class in several contexts so it makes sense *not* to rename it - it has value beyond Profiles and doesn't need to be made profile-specific. For this reason, I recommend closing this Issue.

**Comment by aisaac** [comment]
This is not a receivable argument in the context of a WG document. WG Group Notes can be changed at any time, implementers should be aware of this. 
I am also wary of points like "it has values beyond profiles". We are expected to be relevant for profile requirements, the fact that something would be relevant for other situations is no warrant for ending with constructs that are hard to grasp for us. Actually this points hints that what we need is a subclass of ResourceDescriptor, which would make more sense in the context of profiles. And I'm not sure you want this!
So I'm opposed to closing it. At least on these grounds.

**Comment by nicholascar** [comment]
> what we need is a subclass of ResourceDescriptor, which would make more sense in the context of profiles. And I'm not sure you want this

I have no objection is someone wants to define one, my point is that I can use this current definition for profiles just fine and also for other things. 

No domain is set for the Profile/ResourceDescriptor linking property (`hasResource`) so there's no PROF ontology reason why `hasResource`, and thus ResourceDescriptors themselves, can't be used for anything that wants a ResourceDescriptor.

A ResourceDescriptor is just, as it says, a something that describes a resource in terms of format, conformance, location on the web (artifact) and perhaps role, so it really is generally applicable.

Yes, this is a good argument to do nothing and leave the class named as it is, thus close the issue.

**Comment by aisaac** [comment]
No it's not a valid argument. Minimizing ontological commitment is a great design principle, and one that try to enforce in everything that I'm involved. And for the case at hand, I'm probably going to be ok with the domain of `hasResource` to be left unspecified, in fact.

But ease of re-use in different scenarios shouldn't come at the expense of understandability and usability of the constructs for the original case they're meant for. If only a few understand "Resource Descriptor" then there's an issue that re-usability in a wider context won't solve. As a matter of fact, it is possible that the potential users of the model would be also puzzled by the lack of clarity...
